plugin/declarativenetworkmanager.cpp

```diff
@@ -90,6 +92,11 @@ QString DeclarativeNetworkManager::state() const
     return m_sharedInstance->state();
 }
 
+DeclarativeNetworkManager::State DeclarativeNetworkManager::globalState() const
+{
+    return static_cast<DeclarativeNetworkManager::State>(m_sharedInstance->globalState());
```

Comment:
Random observation: expectation that DeclarativeNetworkManager::State enum matches NetworkManager::State - need for this is documented only at DeclarativeNetworkManager side...

Reply:
Was thinking all the places calling this would have state changed but no harm in explicitly guarding that something changed.

Reply:
Made the enum values copying NetworkManager side so changes only there have no chance of breaking anything.